### PR TITLE
Add requirements, broaden tests, make values more strict

### PR DESCRIPTION
**Pull request by clbarnes**
- Requirements file: make it easier for people to set up a full, fairly replicable dev environment
- Broaden tests: explicitly support some more str-like behaviour
- `__new__` mangling: extend the analogy with IntEnum, which ensures that its values are integers
- `__version__` and `__version_info__`: because I have a compulsion

### CODE REVIEW

File: test_strenum.py [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
+
+import pytest
+
 from strenum import StrEnum
 
 
 class HttpMethod(StrEnum):
     GET = auto()
-    HEAD = auto
+    HEAD = auto()
```
Comment (clbarnes) [inline test_strenum.py]: I think this was a bug? Or at least should have been.

Comment (Elijas) [reply test_strenum.py:10]: Same typo in README.md https://github.com/irgeek/StrEnum/pull/2

Comment (clbarnes) [reply test_strenum.py:10]: Thanks, fixed.

**Comment by irgeek** [comment]
Thanks for the contribution @clbarnes. Sorry it too so long to respond -- I somehow turned off notifications for this repo. I've merged your changes with a couple of changes of my own:

1) I removed `requirements.txt`. The requirements are already described in `setup.py` and I don't want them to drift apart over time. Because of some changes to setuptools, I've reworked the requirements in `setup.py` to make test requirements optional. The README file describes how to install them.

2) I added [versioneer](https://github.com/warner/python-versioneer) to manage the version number -- again so I don't have to manage it in two places. The version information is still available in `strenum.__version__` and `strenum.__version_info__`.

3) I also added the repo to TravisCI!

Thanks again!

**Comment by clbarnes** [comment]
Thanks for getting to it! And those sound like reasonable changes: I am generally in favour of separating setup.py (minimal, flexible runtime dependencies) with requirements.txt (hard version constraints for a replicable fully-provisioned dev environment) but using extras is a fine way to go.

I actually started a discussion on the python-ideas discourse to get more or less this implementation included in the standard library - it's so small and would be so valuable to a lot of downstream projects. But I will be happily recommending the usage of this library alongside, with the next release.

**Pull request merged**